### DISCUSSION TITLE
Randomly Socket Shutdowns/Issues, OSError: [WinError 64]

### DISCUSSION THREAD

**Discussion by Cacsjep** [Kludex/uvicorn#1517] [Potential Issue] [answered]
Hey Guys,

We have running an fastapi asgi app running on uvicorn and sometimes web server is not working because of the following error.
It looks like that socket is closed, this issue happens randomly once a month or so.

I just want to ask if there someting that we can do or is this an open issue, I also create an issue on [cpython](https://github.com/python/cpython/issues/93758) they mean it is uvicorn releated.
Event Viewer on windows does not showing any event when this happpen.
Disabling Windows Defender does not affect or resovle this issue.

I was not able todo a reproducible example.

```
ERROR  Task exception was never retrieved
future: <Task finished name='Task-15680987' coro=<IocpProactor.accept.<locals>.accept_coro() done, defined at C:\Program Files\Python39\lib\asyncio\windows_events.py:568> exception=OSError(22, 'The specified network name is no longer available', None, 64, None)>
Traceback (most recent call last):
  File "C:\Program Files\Python39\lib\asyncio\windows_events.py", line 571, in accept_coro
    await future
  File "C:\Program Files\Python39\lib\asyncio\windows_events.py", line 817, in _poll
    value = callback(transferred, key, ov)
  File "C:\Program Files\Python39\lib\asyncio\windows_events.py", line 560, in finish_accept
    ov.getresult()
OSError: [WinError 64] The specified network name is no longer available

ERROR  Accept failed on a socket
socket: <asyncio.TransportSocket fd=1192, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('0.0.0.0', 8088)>
Traceback (most recent call last):
  File "C:\Program Files\Python39\lib\asyncio\proactor_events.py", line 819, in loop
    conn, addr = f.result()
  File "C:\Program Files\Python39\lib\asyncio\windows_events.py", line 571, in accept_coro
    await future
  File "C:\Program Files\Python39\lib\asyncio\windows_events.py", line 817, in _poll
    value = callback(transferred, key, ov)
  File "C:\Program Files\Python39\lib\asyncio\windows_events.py", line 560, in finish_accept
    ov.getresult()
OSError: [WinError 64] The specified network name is no longer available
```
OS: Windows 10 Pro
Python: 3.9.13

**Comment by Cacsjep** [comment, marked as answer]
Is not unicorn releated

**Reply by doctor-entropy** [reply]
@Cacsjep Can you give us more details about your answer. I am facing this exact same issue on Windows Server 2016 and I am not sure what to do about this. Any help is greatly appreciated.

**Reply by Cacsjep** [reply]
look into my [issue](https://github.com/python/cpython/issues/93821#issuecomment-1621454710 ) on cpython to dive in =)

**Reply by doctor-entropy** [reply]
@Cacsjep Thanks for the help! My application actually crashes randomly on a request. So would your solution process the request? Or would it only catch the error and prevent the entire application from crashing (and needing restart).

**Reply by Cacsjep** [reply]
This post is long time ago when i had this issue, if its correct in my head, then the issue is that a request is made and closed in a very short time, lets say your client do a request and a very short after network is loss or so. we had this trouble with uploads from LTE routers. The request is unable to proccess because the connection is reseted or unable to be accepted. so it prevents app from crashing.

**Reply by doctor-entropy** [reply]
@Cacsjep Although your solution doesn't prevent the whole application from crashing (which is really nice), The cause of my issue was something more silly. I was executing the Python web server from Powershell and unbeknownst to me, Powershell had the "QuickEdit" and "Insert" Mode turned on which was causing my application to freeze every time I clicked on the Powershell window. Which was causing network-related issues.

Turning the QuickEdit and Insert mode off worked for me and I don't get that WinError 64 issue anymore.

https://serverfault.com/questions/204150/sometimes-powershell-stops-sending-output-until-i-press-enter-why

**Comment by stone9693** [comment]
suggest use powershell with rights of administrator
